### ISSUE TITLE
round to units

### ISSUE THREAD

**Issue by rafaqz** [PainterQubits/Unitful.jl#207]
While single argument `round(1.5m)` and `trunc(3.4s)` etc can give unpredictable results, they wouldn't if we always specified the dimension with the type to round to using the two argument syntax :

```julia
using Unitful
using Unitful: m, cm
import Base: round
round(::Type{T}, x::Quantity) where {T<:Quantity} = T(round(typeof(zero(T).val), uconvert(unit(T), x).val))

julia> round(typeof(1m), 50.78cm)
1 m
```

Although there is probably a cleaner way to write that. 

I'm finding round() to be needed all the time for plotting series of environmental data. Being able to just round to the start of a day is very useful when using evenly spaced months or years where days need to be fractional.

Edit: I'll put this into a pull request if its useful.

**Comment by c42f** [comment]
This seems sensible to me; it's basically the same thing as #205 / #48 but for a different function.

Related to plotting, did you see `UnitfulPlots.jl`?

**Comment by rafaqz** [comment]
Cool. I've seen UnitfulPlots, most of the commits in the last year are mine :)

But it needs some love too. I've got another pull request sitting around for it, as it doesn't handle functions or DiffEq output. Function are an easy fix but debugging DiffEq output is awful.

**Comment by rafaqz** [comment]
I have a pull request ready to go for this (and trunc etc) but there's already a backlog of my pull requests in the queue :/

**Comment by ajkeller34** [comment]
Sorry, other work is unusually overwhelming right now; I will try to review PRs soon. Thanks for your patience.

**Comment by ajkeller34** [comment]
A rounding PR came up again (https://github.com/PainterQubits/Unitful.jl/pull/246) and I wonder if this might be the time to revisit this issue. @rafaqz did you have a branch on your fork for this?

**Comment by rafaqz** [comment]
PR is in.

**Comment by ajkeller34** [comment]
Closed by #249, #246
